### ISSUE TITLE
toggleFullScreenMode() gives error FlutterError (Navigator operation requested with a context that does not include a Navigator

### ISSUE THREAD

**Issue by jimmyff** [sarbagyastha/youtube_player_flutter#150]
When I attempt to make the video full screen I get the following error:
```
FlutterError (Navigator operation requested with a context that does not include a Navigator.
The context used to push or pop routes from the Navigator must be that of a widget that is a descendant of a Navigator widget.)
```

I'm instegating this from a button using 
```
onPressed: () => _controller.toggleFullScreenMode();
```

**Comment by jimmyff** [comment]
I have looked in to this problem. This is the most useful explanation I've found:
https://stackoverflow.com/questions/44004451/navigator-operation-requested-with-a-context-that-does-not-include-a-navigator

I am utilising the `builder` parameter of `MaterialApp` which says in the documentation `Material specific features such as [showDialog] and [showMenu], and widgets such as [Tooltip], [PopupMenuButton], also require a [Navigator] to properly function.` -so I do not understand why this isnt working?

I've also attempted to the `YouTubePlayer` widget in a builder however that didnt work.

This is my Material App:
```dart
    return new StoreProvider<AppState>(
        store: widget.store,
        child: MaterialApp(
          title: 'Media Flutter App',
          builder: (context, child) {
            return StoreConnector<AppState, MediaPanelViewModel>(
                converter: (store) => MediaPanelViewModel.fromStore(store),
                builder: (context, viewModel) {
                  return MediaPlayerScaffold(
                    drawer: Drawer(),
                    bottomNavigationBar: viewModel.currentlyPlaying == null
                        ? null
                        : MediaPanel(),
                    body: child,
                  );
                });
          },
          initialRoute: HomeRoute,
          onGenerateRoute: generateRoute,
        ));
```

`MediaPanel` is where I utilise `YoutubePlayer`.

Any ideas how I can fix it? I've looked through your code to see if I can find where it needs to utilise the Navigator but cant find it.

**Comment by sarbagyastha** [comment]
Cloud you please provide a reproducible example ?

**Comment by jimmyff** [comment]
yep, leave it with me

**Comment by jimmyff** [comment]
Hi @sarbagyastha, I've created an little project to highlight the issue:
[bug.zip](https://github.com/sarbagyastha/youtube_player_flutter/files/4252257/bug.zip)

To reproduce the full screen error:

- Tap 'play video 1'
- Tap full screen icon on the media bar at the bottom

_**Aside**: This project also demonstrates a second issue, if you click on the playing video thumbnail I wanted to open it bigger at the bottom however the youtube player seems to break, I'm not sure what I'm doing wrong - maybe the webview is getting recreated. It would be nice to resolve this issue too but the full screen video issue is my main priority._ 

Any help getting the full screen working would be massively appreciated.

Thanks very much,
Jimmy

**Comment by jimmyff** [comment]
Hey @sarbagyastha I think having the option of passing in a global navigator key would fix the fullscreen issue?

https://stackoverflow.com/questions/52962112/how-to-navigate-without-context-in-flutter-app/53397266#53397266

https://github.com/brianegan/flutter_redux/issues/5#issuecomment-361215074

What do you think?

**Comment by jimmyff** [comment]
Hey @sarbagyastha did you manage to consider my suggestion for adding a navigator key parameter? I can implement it and send a PR if you don't have chance as this blocking me launching my app. 

I would just make sure that wherever you are utilising Navigator it would use the key if provided. place such as here: https://github.com/sarbagyastha/youtube_player_flutter/blob/e6e04d655d5ca285aac4b4a9d32c881af2eb1a40/lib/src/player/youtube_player.dart#L231

**Comment by jimmyff** [comment]
Created a PR that fixes this issue. Could you have a look?

https://github.com/sarbagyastha/youtube_player_flutter/pull/160

Thanks

**Comment by sarbagyastha** [comment]
Should fix with v7.0.0. Feel free to reopen if the issue persists.
